src/generator/AutoRest.Ruby.Azure/TemplateModels/AzureMethodTemplateModel.cs

```diff
@@ -273,7 +273,8 @@ public override string OperationReturnTypeString
         {
             get
             {
-                if (Extensions.ContainsKey("nextMethodName") && !Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                if (Extensions.ContainsKey("nextMethodName") && (!Extensions.ContainsKey(AzureExtensions.PageableExtension) ||
+                    (Extensions.ContainsKey(AzureExtensions.PageableExtension) && Extensions.ContainsKey(AzureExtensions.LongRunningExtension))))
```

Comment:
there's a parenthesis enclosing the last two conditions, so this is needed.

---
In reply to: [74297529](https://github.com/Azure/autorest/pull/1335#discussion_r74297529) [](ancestors = 74297529)